src/pages/Lend/LoanSearchPage.vue

```diff
@@ -138,6 +196,40 @@ export default {
 			FLSS_ONGOING_EXP_KEY,
 			'EXP-VUE-FLSS-Ongoing-Sitewide'
 		);
+
+		// Extended FLSS Loan Filter Experiment
+		const challengeHeaderExpData = this.apollo.readFragment({
+			id: `Experiment:${CHALLENGE_HEADER_EXP}`,
+			fragment: experimentVersionFragment,
+		}) || {};
+		this.enableChallengeHeader = challengeHeaderExpData?.version === 'b';
+
+		if (this.enableChallengeHeader) {
+			const teamPublicId = this.$route?.query?.team ?? '';
+			let teamId = null;
+			if (teamPublicId) {
+				const teamInfo = this.apollo.readQuery({ query: TeamInfoFromId, variables: { team_public_id: teamPublicId } }); // eslint-disable-line max-len
+				teamId = teamInfo?.community?.team?.id ?? null;
+			} else {
+				const userTeamsData = this.apollo.readQuery({ query: myTeamsQuery });
+				const userTeams = userTeamsData.my?.teams?.values ?? [];
+
+				let maxAmountLentTeam = {};
+				if (userTeams.length > 0) {
+					maxAmountLentTeam = userTeams?.reduce((prev, current) => {
```

Comment:
This `reduce` could be moved to a function outside of the component in this file and shared between `preFetch` and `created`